dask/dataframe/optimize.py

```diff
@@ -17,12 +17,16 @@ def optimize(dsk, keys, **kwargs):
 
     if not isinstance(dsk, HighLevelGraph):
         dsk = HighLevelGraph.from_collections(id(dsk), dsk, dependencies=())
-
-    dsk = optimize_dataframe_getitem(dsk, keys=keys)
-    dsk = optimize_blockwise(dsk, keys=keys)
-    dsk = fuse_roots(dsk, keys=keys)
+    else:
+        # Perform Blockwise optimizations for HLG input
+        dsk = optimize_dataframe_getitem(dsk, keys=keys)
+        dsk = optimize_blockwise(dsk, keys=keys)
+        dsk = fuse_roots(dsk, keys=keys)
```

Comment:
Just for my own understanding, we're adding the `else` block here because none of these optimizations will apply to a HLG which is only `MaterializedLayer`s?

Reply:
Yes - Thanks for calling this out.  My understadning is that these optimizations only make sense for Blockwise Layers.